src/components/icon-button/IconButton.root.tsx

```diff
@@ -3,29 +3,38 @@ import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components
 
 import { setAlpha, Utils } from '../../shared';
 import { TTheme } from '../../utilities/theme';
-import Spacing from '../../utilities/spacing';
+import Spacing, { SPACING_TOKENS } from '../../utilities/spacing';
 import { applyMargin, applyPadding } from '../../utilities/layout';
 import { applyShape } from '../../foundations/shape';
 import { applyTextStyles } from '../text';
 
-import { DEFAULT_ICON_BUTTON_SIZE, ICON_BUTTON_DEFINITIONS } from './IconButton.constants';
+import { ICON_BUTTON_DEFINITIONS } from './IconButton.constants';
 import { PIconButtonRoot } from './IconButton.props';
 
 const IconButtonRoot = styled.button.withConfig<PIconButtonRoot>({
     shouldForwardProp: (property, validator) =>
         Utils.blockProperty(property) && validator(property),
 })(
     ({
-        size = DEFAULT_ICON_BUTTON_SIZE,
-        inverted = false,
-        toggled = false,
-        destructive = false,
-        disabled = false,
-        theme: { palette, action, text },
+        size,
+        compact,
+        inverted,
+        toggled,
+        destructive,
+        disabled,
+        theme: { palette, action, text, animation },
     }: ThemedStyledProps<PIconButtonRoot, TTheme>): FlattenSimpleInterpolation => {
         const isDefault = !inverted && !destructive && !toggled;
         const { main, contrast } = destructive && !toggled ? palette.alert : palette.primary;
 
+        // this solution is not ideal and should probably be refactored in the
+        // `applyPadding` and `applyMargin` mixin-function
+        let spacingTokenIndex = SPACING_TOKENS.indexOf(ICON_BUTTON_DEFINITIONS[size].spacing);
```

Comment:
nit: I am not sure its a good idea to use this expensive operation and it takes some time to figure out whats going on. Isn't it better to have a separate spacing prop for the compact versions in the definitions map?

Reply:
probably ... as mentioned in the comment this is not ideal and should be rethought

Reply:
any ideas?

Reply:
I was trying to find a way to mutate the array with one of the sorting methods but I think the simplest solution might be something like: 
 
 ```
const ICON_BUTTON_DEFINITIONS: { [size in TIconButtonSizeToken]: TIconButtonDefinition } = {
    xs: {
       spacingCompact: 0,
        spacing: 75,
        iconSize: 12,
        fontSize: 75,
    },
    ...
  }
 ```
 
 If I understand the requirement correctly? We don't need to move the index and we can just take the value directly

Reply:
@nevyangelova I did apply your approach for now, but once we get to have more components that have a `compact` version (candidates can potentially be `Button`, `TextField` or `Checkbox` etc.) we should rethink the probability of adding it to the layout mixins (`applyMargin` and `applyPadding`)